### DISCUSSION TITLE
Improve setting amount when adding a donation line item in the CP

### DISCUSSION THREAD

**Discussion by icreatestuff** [craftcms/commerce#4087] [Ideas]
### Description
When adding a donation line item to an order from the control panel there is no option to set the donation amount, which then triggers an error preventing the order from being able to update on save.

Curiously you can set the quantity of donation line items to add, which presumably should only ever be a maximum of 1.

### Steps to reproduce
1. Enable donations
2. Create an order
3. Add the donation line item
4. See the error about no Donation Amount set

### Expected behavior
Adding a donation line item should allow setting the amount

### Actual behavior
No amount field is visible when adding the donation line item, causing the amount to be set to 0 and therefore causing an error

### Craft CMS version

Pro 4.13.0

### Craft Commerce version

4.7.0

### PHP version

8.1.29

### Installed plugins and versions

-

**Comment by linear** [comment]
<p><a href="https://linear.app/craftcms/issue/PT-2322/[4x]-adding-a-donation-line-item-via-control-panel-doesnt-allow">PT-2322 [4.x]: Adding a donation line item via control panel doesn't allow setting amount</a></p>

**Comment by nfourtythree** [comment]
Hi @icreatestuff 

Thank you for your feedback.

It is possible to create an amount on a donation line item. It follows the same convention, technically, as it does when creating a donation from the front end. This is in the fact that it uses the `donationAmount` key in the line item options.

The video below shows how this is possible:

https://github.com/user-attachments/assets/8f436553-523a-4da0-ada4-75d74f661932

We understand that this is not the most ideal way of managing this process. I will update this issue to be a feature request so this is something that we can look into improving in a future release of Commerce.

Thanks!

**Comment by icreatestuff** [comment]
Thanks @nfourtythree for the explanation and demo video. As a dev, having some understanding of the underlying architecture for how Commerce is working, I could just about work with that. For clients, this is too complex a process. Thanks for making this a feature request.
